src/main/java/org/quiltmc/loader/impl/discovery/ClasspathModCandidateFinder.java

```diff
@@ -46,15 +46,10 @@ public void findCandidates(QuiltLoaderImpl loader, BiConsumer<URL, Boolean> appe
 		if (QuiltLauncherBase.getLauncher().isDevelopment()) {
 			// Search for URLs which point to 'fabric.mod.json' entries, to be considered as mods.
 			try {
-				Enumeration<URL> mods = QuiltLauncherBase.getLauncher().getTargetClassLoader().getResources("fabric.mod.json");
 				Set<URL> modsList = new HashSet<>();
-				while (mods.hasMoreElements()) {
-					try {
-						modsList.add(UrlUtil.getSource("fabric.mod.json", mods.nextElement()));
-					} catch (UrlConversionException e) {
-						loader.getLogger().debug(e);
-					}
-				}
+
+				addModSources(loader, modsList, "quilt.mod.json");
+				addModSources(loader, modsList, "fabric.mod.json");
```

Comment:
could this result in double adding of mods which contain both files?

Reply:
I don't think so, since it has a `Set<URL>`, so it will call `URL.hashCode()` and `URL.equals()` to check that they are same. (And the URL's are of the *base* resource, not the `*.mod.json` file so comparing them will be useful).

Wait a minute, isn't `URL.equals` meant to be avoided at all costs? If so then this needs re-writing anyway to avoid that.